test/evm_test.exs

```diff
@@ -1,8 +1,68 @@
-defmodule EVMTest do
-  use ExUnit.Case
-  doctest EVM
+defmodule EvmTest do
+  use ExUnit.Case, async: true
 
-  test "the truth" do
-    assert 1 + 1 == 2
+  @passing_tests %{
+    arithmetic: [
+      :add0,
+      :add1,
+    ]
+  }
+
+
+  test "Ethereum Common Tests" do
+    for {test_type, test_group} <- @passing_tests do
+      for {test_name, test} <- read_test_file(test_type),
+        Enum.member?(test_group, String.to_atom(test_name)) do
+        db = MerklePatriciaTree.Test.random_ets_db()
+        state = EVM.VM.run(
+          MerklePatriciaTree.Trie.new(db),
+          hex_to_int(test["exec"]["gas"]),
+          %EVM.ExecEnv{
+            machine_code: hex_to_binary(test["exec"]["code"]),
+          }
+        )
+
+        assert_state(test, state)
+        assert elem(state, 1) == hex_to_int(test["gas"])
+      end
+    end
+  end
+
+  def read_test_file(type) do
+    {:ok, body} = File.read(test_file_name(type))
+    Poison.decode!(body)
+  end
+
+  def test_file_name(type) do
+    "test/support/ethereum_common_tests/VMTests/vm#{Macro.camelize(Atom.to_string(type))}Test.json"
   end
+
+  def hex_to_binary(string) do
```

Comment:
I tried putting these in `test_helper.exs` but it complained that they wanted to be in a module. Do I need to use macros for helper functions in tests?